axiom/nr_txn.c

```diff
@@ -1638,6 +1641,109 @@ void nr_txn_record_error(nrtxn_t* txn,
   nr_free(span_id);
 }
 
+void nr_txn_record_error_with_additional_attributes(
```

Comment:
There is a large amount of code duplication in this API. It would be nice to create a more unified API such that we don't have multiple copies of large functions.

Reply:
Good point, I have added helper functions for `nr_txn_record_error`, `nr_segment_set_error`, and `nr_error_create` to reduce code duplication: [23823f9](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/23823f9ac52055e35d00951a9af83aadaeb756fe) and [f917ced](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/f917ced797ecc2b0f2f560487081e73bf4cf47a1)

Reply:
Could you remove `nr_txn_record_error_helper` and then just have `nr_txn_record_error` call `nr_txn_record_error_with_additional_attributes` with the unused fields filled with nothing values?

Reply:
Good idea, I have removed all of the helper functions: [e177f40](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/e177f40a8b868410e5e1b728df7b5df8356ce1f4), [ed191c0](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/ed191c0b4ee2cff587aa3fcee1219c9133e586f7), [788c5a0](https://github.com/newrelic/newrelic-php-agent/pull/942/commits/788c5a0e6330c7c788b2a6c5b028391319b7af17)